### PR TITLE
docs: added definition of json hyper-schema

### PR DESCRIPTION
**Pull request by Akshaybagai52**
**GitHub Issue:** # / NA

**Summary**:

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

Yes/No
[JUSTIFICATION]

### CODE REVIEW

File: pages/learn/glossary.md [hunk 1]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
 The JSON format is an open format, with its own [homepage](https://www.json.org/), and specifications published in the [ECMA-404](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) and [RFC-8259](https://datatracker.ietf.org/doc/html/rfc8259) documents from ECMA and the IETF respectively.
 In particular, it is not managed or developed by the JSON Schema team, who simply make use of the format.
 
+### JSON Hyper-Schema
+
+JSON Hyper-Schema extends JSON Schema, offering a vocabulary to annotate JSON documents with hypermedia controls. This extension facilitates the description of links and actions that can be executed on JSON data, making it a potent tool for developing hypermedia-driven APIs.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:52]: This is a great opening.

Let's add a link on "vocabulary" to that definition in the glossary.

File: pages/learn/glossary.md [hunk 2]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+Built upon the foundation of JSON Schema, JSON Hyper-Schema introduces additional keywords to specify hyperlinks and instructions for handling remote JSON resources in hypermedia environments like HTTP. The Link Description Object (LDO) is a core concept, encapsulating information about available actions. LDO properties include "rel" (relation type), "href" (URI template), and "templateRequired" (required template variables).
+
+The "href" keyword in JSON Hyper-Schema is crucial, allowing the description of URLs using flexible URI templates with variables resolved against instance data. Concepts like "base" establish the URI base for references, aiding in constructing relative URLs. The "links" property associates Link Description Objects with instances, enabling the definition of multiple links and their relations.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:56]: I think this is the wrong level of detail for a glossary entry and not very helpful. I think it would be better to not describe specific mechanisms, but rather describe behaviors, experiences, and outcomes. You can mention that hypermedia controls are defined using the `links` keyword, but I wouldn't go as far as describing LDOs here. Focus on _what_ they can do rather than _how_ they do it.

For example, I think this should describe how the hypermedia controls defined in `links` are templates that are actualized by applying an instance to the URI Templates in the description. A JSON instance might have an "author" property whose value is the identifier of an author. The hyper-schema that describes that instance might have an LDO that uses the id in that property to construct a link to an instance for the author.

File: pages/learn/glossary.md [hunk 3]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+In conclusion, JSON Hyper-Schema enhances JSON Schema, introducing a vocabulary for hypermedia controls and enabling the creation of more interactive APIs. It provides a structured approach to articulate links and actions on JSON documents, enhancing client understanding and interaction with remote JSON resources.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:58]: "In conclusion" is a little formal sounding for this kind of content.

JSON Hyper-Schema is good for adding hypermedia controls to an existing JSON HTTP API in a non-breaking way. That might be a good thing to mention.

Comment (Akshaybagai52) [reply pages/learn/glossary.md:58]: Thanks @jdesrosiers for this descriptive and informative review I will modify my PR accordingly

**Comment by Akshaybagai52** [comment]
@jdesrosiers can you review this now
Thanks

File: pages/learn/glossary.md [hunk 4]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+The essence of JSON Hyper-Schema lies in its ability to define links and actions that can be executed on JSON data. This is achieved through the use of the links keyword, which allows for the creation of dynamic, interactive data representations. For example, a JSON document representing a blog post might include an "author" property. The JSON Hyper-Schema that describes this document could include a hypermedia control that uses the author's identifier to construct a link to the author's profile. This link is a template that, when applied with the actual data, provides direct access to the author's profile, enhancing the user experience by offering a seamless navigation experience.
+
+This approach not only simplifies the interaction with JSON data but also enhances the overall user experience by providing direct access to related resources without the need for additional requests. It allows developers to create rich, interactive JSON documents that can dynamically adapt to the data they represent, making JSON Hyper-Schema a potent tool for developing hypermedia-driven APIs.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:56]: > without the need for additional requests.

This might be misleading. Technically, there are additional HTTP requests. The difference is just that you don't have to construct those additional requests, they can be done for you.

Comment (jdesrosiers) [inline pages/learn/glossary.md:56]: > that can dynamically adapt to the data they represent

I don't understand what you mean by this.

File: pages/learn/glossary.md [hunk 5]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+To sum up, JSON Hyper-Schema enhances JSON Schema by introducing features tailored for hypermedia controls. This facilitates the creation of more interactive APIs without disrupting existing JSON HTTP APIs. It adds a layer of interactivity to JSON documents, making it easier for clients to understand and interact with remote JSON resources.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:58]: A couple editorial suggestions.

```suggestion
To sum up, JSON Hyper-Schema enhances JSON Schema by introducing features for creating hypermedia controls. This facilitates the creation of more interactive APIs without disrupting existing JSON HTTP APIs. It adds a layer of interactivity to JSON documents, making it easier to interact with remote JSON resources.
```

Comment (jdesrosiers) [inline pages/learn/glossary.md:58]: > without disrupting existing JSON HTTP APIs

I don't think people are going to understand what you mean by this. Either you need to elaborate a little more, or leave this bit out.

**Review by jdesrosiers — COMMENTED** [review]
I left some comments. Progress looks good.

Btw, you seem to have pushed a commit that doesn't belong here. Perhaps it was meant for a different branch? Please make sure the clean that up.

@benjagm Aside from the technical aspects @Akshaybagai52 and I are working through, what do you think about the voice and tone so far? It strikes me as a bit wordy and salesy (like someone is trying to sell me something) than I would expect from a glossary entry.

**Comment by Akshaybagai52** [comment]
@jdesrosiers Thanks for descriptive review. Sorry for the different code push I forgot that my PR is still open for same branch. I will remove the code and will work on your suggestions
Thanks

File: pages/learn/glossary.md [hunk 6]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
 The JSON format is an open format, with its own [homepage](https://www.json.org/), and specifications published in the [ECMA-404](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) and [RFC-8259](https://datatracker.ietf.org/doc/html/rfc8259) documents from ECMA and the IETF respectively.
 In particular, it is not managed or developed by the JSON Schema team, who simply make use of the format.
 
+### JSON Hyper-Schema
+
+JSON Hyper-Schema extends JSON Schema, offering a [vocabulary](https://json-schema.org/draft/2019-09/json-schema-hypermedia) to annotate JSON documents with hypermedia controls. This extension facilitates the description of links and actions that can be executed on JSON data, making it a potent tool for developing hypermedia-driven APIs.
```
[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply pages/learn/glossary.md:52]: done

File: pages/learn/glossary.md [hunk 7]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+The essence of JSON Hyper-Schema lies in its ability to define links and actions that can be executed on JSON data. This is achieved through the use of the links keyword, which allows for the creation of dynamic, interactive data representations. For example, a JSON document representing a blog post might include an "author" property. The JSON Hyper-Schema that describes this document could include a hypermedia control that uses the author's identifier to construct a link to the author's profile. This link is a template that, when applied with the actual data, provides direct access to the author's profile, enhancing the user experience by offering a seamless navigation experience.
```
[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply pages/learn/glossary.md:54]: done

[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply pages/learn/glossary.md:54]: done

[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply pages/learn/glossary.md:54]: I changed "user" to "developer"

File: pages/learn/glossary.md (hunk 4 above)
Comment (Akshaybagai52) [reply pages/learn/glossary.md:56]: I changed "without the need for additional requests." to "with a streamlined process that automates additional HTTP requests."

Comment (Akshaybagai52) [reply pages/learn/glossary.md:56]: Do I change 
"it allows developers to create rich, interactive JSON documents that can dynamically adapt to the data they represent," 
to this :
"Developers can leverage this capability to create dynamic and interactive JSON documents, allowing them to adapt seamlessly to the evolving nature of the data they represent."

It means : JSON documents can adapt dynamically to changes in the data they represent,

File: pages/learn/glossary.md (hunk 5 above)
Comment (Akshaybagai52) [reply pages/learn/glossary.md:58]: changed content to this :
"This allows for the development of more interactive APIs while preserving the integrity of existing JSON HTTP APIs."

**Comment by Akshaybagai52** [comment]
@jdesrosiers I've reverted the code and did further changes in documentation

File: pages/learn/glossary.md (hunk 4 above)
Comment (Akshaybagai52) [reply pages/learn/glossary.md]: > Do I change "it allows developers to create rich, interactive JSON documents that can dynamically adapt to the data they represent," to this : "Developers can leverage this capability to create dynamic and interactive JSON documents, allowing them to adapt seamlessly to the evolving nature of the data they represent."
> 
> It means : JSON documents can adapt dynamically to changes in the data they represent,

I will update this according to your review

File: pages/learn/glossary.md [hunk 8]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
 In particular, it is not managed or developed by the JSON Schema team, who simply make use of the format.
 
+### JSON Hyper-Schema
+
+JSON Hyper-Schema extends JSON Schema, offering a [vocabulary](#vocabulary) to annotate JSON documents with hypermedia controls. This extension facilitates the description of links and actions that can be executed on JSON data, making it a potent tool for developing hypermedia-driven APIs.
+
+The essence of JSON Hyper-Schema lies in its ability to define links and actions that can be executed on JSON data. This is achieved through the use of the `links` keyword, which allows for the creation of dynamic, interactive data representations. For example, a JSON document representing a blog post might include an "author" property. The JSON Hyper-Schema that describes this document could include a template for a hypermedia control that uses the author's identifier in the actual data to construct a link to the author's profile. This link is a template that, when applied with the actual data, provides direct access to the author's profile, enhancing the developer experience by offering a seamless navigation experience.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:54]: This still contains the same confusion. The link is not a template. The link is the result of applying the template, which you describe adequately in the previous sentence.

Here are my suggestions.

```suggestion
The essence of JSON Hyper-Schema lies in its ability to define links and actions that can be executed on JSON data. This is achieved through the use of the `links` keyword, which allows for the creation of dynamic, interactive data representations. For example, a JSON document representing a blog post might include an "author" property. The JSON Hyper-Schema that describes this document could include a template for a hypermedia control that uses the author's identifier in the [instance](#instance) to construct a link to the author's profile. The developer doesn't need to construct the URL manually, which enhances the developer experience by offering a seamless navigation experience.
```

File: pages/learn/glossary.md [hunk 9]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+This approach not only simplifies the interaction with JSON data but also enhances the overall developer experience by providing direct access to related resources with a streamlined process that automates additional HTTP requests. It allows developers to create rich, interactive JSON documents that can dynamically adapt to the data they represent, making JSON Hyper-Schema a potent tool for developing hypermedia-driven APIs.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:56]: I still don't think this makes sense. I suggest just dropping this paragraph.

File: pages/learn/glossary.md [hunk 10]
```diff
@@ -47,6 +47,16 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+To sum up, JSON Hyper-Schema enhances JSON Schema by introducing features for creating hypermedia controls. This allows for the development of more interactive APIs while preserving the integrity of existing JSON HTTP APIs. It adds a layer of interactivity to JSON documents, making it easier to interact with remote JSON resources.
```
Comment (jdesrosiers) [inline pages/learn/glossary.md:58]: > This allows for the development of more interactive APIs while preserving the integrity of existing JSON HTTP APIs.

You used different words, but you still have the same problem. You're making a vague statement that you don't explain in any way. I suggest replacing the previous paragraph with an explanation of this concept.

Imagine the scenario where you have an existing API that serves JSON and you'd like to introduce some hypermedia concepts. One option is to change your API to use a hypermedia enabled media type like HAL, but that would mean changing your whole API to use a different media type. All of the consumers of your API would need to change their code to process HAL rather than plain JSON.

But, with JSON Hyper-Schema, you can add those hypermedia controls to your API without changing the JSON that you API returns. The schemas are associated with the JSON response through an HTTP Link header. Clients can construct the hypermedia controls from that schema and the JSON response, but they can also ignore the schema and use the plain JSON. That means that linking a hyper-schema to a response is a backward compatible change. Existing code consuming the API will still work the way it always has. Consumers can start using the hyper-schema that came with the response when they're ready.

This also means that the API doesn't need to be changed completely over to using hypermedia all at once. You can add hypermedia controls a little at a time.

The hard part for you is going to be writing something _brief_ that gets that concept across without going into too many details.

**Review by jdesrosiers — COMMENTED** [review]
I left a few more notes.

I looks like some of you're extraneous commit didn't get completely reverted. I suggest learning how to drop a commit rather than reverting the change with another commit.

**Comment by benjagm** [comment]
> Aside from the technical aspects @Akshaybagai52 and I are working through, what do you think about the voice and tone so far? It strikes me as a bit wordy and salesy (like someone is trying to sell me something) than I would expect from a glossary entry.

Agreed. Let's review the tone to be just descriptive as expected in glossary instead of convincing ... selling.

**Comment by Akshaybagai52** [comment]
@jdesrosiers can you check now

**Review by jdesrosiers — APPROVED** [review]
Thanks for sticking through this process! @benjagm may make additional suggestions for tone/voice/formatting/etc, but it's good to go from a technical perspective.

**Comment by Akshaybagai52** [comment]
> Thanks for sticking through this process! @benjagm may make additional suggestions for tone/voice/formatting/etc, but it's good to go from a technical perspective.

Thankyou for all the detailed explanation on each commit. learned alot from this

File: pages/overview/[slug].page.tsx [hunk 11]
```diff
@@ -13,7 +13,6 @@ export async function getStaticProps(args: any) { return getStaticMarkdownProps(
 
 export default function StaticMarkdownPage ({ frontmatter, content }: { frontmatter: any, content: any }) {
   const newTitle = 'JSON Schema - ' + frontmatter.title
-
```
Comment (benjagm) [inline pages/overview/[slug].page.tsx:16]: Why this change? Let's discard it.

File: components/Sidebar.tsx [hunk 12]
```diff
@@ -318,4 +318,4 @@ export const DocsNav = () => {
 }
 
 export const getLayout = (page: React.ReactNode) =>
-  getSiteLayout(<SidebarLayout>{page}</SidebarLayout>)
+  getSiteLayout(<SidebarLayout>{page}</SidebarLayout>)
```
Comment (benjagm) [inline components/Sidebar.tsx:321]: Why this change? Let's discard it.

File: pages/learn/glossary.md [hunk 13]
```diff
@@ -47,6 +47,14 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
 The JSON format is an open format, with its own [homepage](https://www.json.org/), and specifications published in the [ECMA-404](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) and [RFC-8259](https://datatracker.ietf.org/doc/html/rfc8259) documents from ECMA and the IETF respectively.
 In particular, it is not managed or developed by the JSON Schema team, who simply make use of the format.
 
+### JSON Hyper-Schema
+
+JSON Hyper-Schema extends JSON Schema, offering a [vocabulary](#vocabulary) to annotate JSON documents with hypermedia controls. This extension facilitates the description of links and actions that can be executed on JSON data, making it a potent tool for developing hypermedia-driven APIs.
```
Comment (benjagm) [inline pages/learn/glossary.md:52]:
```suggestion
JSON Hyper-Schema extends JSON Schema, offering a [vocabulary](#vocabulary) to annotate JSON documents with hypermedia controls. This extension facilitates the description of links and actions that can be executed on JSON data, making it a powerful tool for developing hypermedia-driven APIs.
```

File: pages/learn/glossary.md [hunk 14]
```diff
@@ -47,6 +47,14 @@ The two are, however, entirely different pieces of the conceptual puzzle, with J
...
+
+The essence of JSON Hyper-Schema lies in its ability to define links and actions that can be executed on JSON data. This is achieved through the use of the `links` keyword, which allows for the creation of dynamic, interactive data representations. For example, a JSON document representing a blog post might include an "author" property. The JSON Hyper-Schema that describes this document could include a template for a hypermedia control that uses the author's identifier in the [instance](#instance) to construct a link to the author's profile. The developer doesn't need to construct the URL manually, which enhances the developer experience by offering a seamless navigation experience.
+
+To sum up, JSON Hyper-Schema enhances JSON Schema by introducing features for creating hypermedia controls. This facilitates the creation of more interactive APIs and ensures compatibility with existing JSON HTTP APIs, maintaining a seamless integration. It adds a layer of interactivity to JSON documents, making it easier to interact with remote JSON resources.
```
Comment (benjagm) [inline pages/learn/glossary.md:56]:
```suggestion
In other words, JSON Hyper-Schema extends JSON Schema by introducing features for creating hypermedia controls. This facilitates the creation of interactive APIs and ensures compatibility with existing JSON HTTP APIs, maintaining a seamless integration. It adds a layer of interactivity to JSON documents, making it easier to interact with remote JSON resources.
```

**Review by benjagm — CHANGES_REQUESTED** [review]
This is a top entry for the glossary! Thanks for the hard work.

Left 2 small suggestions and 2 comments.

**Comment by Akshaybagai52** [comment]
continue in this PR https://github.com/json-schema-org/website/pull/405
